### PR TITLE
Various mappings in net/minecraft/world/entity

### PR DESCRIPTION
**Pull request by PiTheGuy**

### CODE REVIEW

File: data/net/minecraft/world/entity/AgeableMob.mapping [hunk 1]
```diff
@@ -0,0 +1,3 @@
+CLASS net/minecraft/world/entity/AgeableMob
+	METHOD getBreedOffspring (Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/entity/AgeableMob;)Lnet/minecraft/world/entity/AgeableMob;
+		ARG 1 level
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/AgeableMob.mapping:3]: This is correct, the other argument should probably be "otherParent"

File: data/net/minecraft/world/entity/ai/behavior/AcquirePoi.mapping [hunk 2]
```diff
@@ -1,8 +1,29 @@
 CLASS net/minecraft/world/entity/ai/behavior/AcquirePoi
+	METHOD <init> (Lnet/minecraft/world/entity/ai/village/poi/PoiType;Lnet/minecraft/world/entity/ai/memory/MemoryModuleType;Lnet/minecraft/world/entity/ai/memory/MemoryModuleType;ZLjava/util/Optional;)V
+		ARG 1 poiType
+		ARG 3 memoryToAcquire
+		ARG 4 onlyIfAdult
+		ARG 5 onPoiAcquistutionEvent
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/AcquirePoi.mapping]: ``onPoiAcquistutionEvent`` -> ``onPoiAcquisitionEvent``

File: data/net/minecraft/world/entity/ai/behavior/AcquirePoi.mapping [hunk 3]
```diff
@@ -1,8 +1,29 @@
...
+	METHOD <init> (Lnet/minecraft/world/entity/ai/village/poi/PoiType;Lnet/minecraft/world/entity/ai/memory/MemoryModuleType;ZLjava/util/Optional;)V
+		ARG 1 poiType
+		ARG 3 onlyIfAdult
+		ARG 4 onPoiAcquisitionEvent
 	METHOD checkExtraStartConditions (Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/entity/PathfinderMob;)Z
 		ARG 1 level
 		ARG 2 owner
+	METHOD constructEntryConditionMap (Lnet/minecraft/world/entity/ai/memory/MemoryModuleType;Lnet/minecraft/world/entity/ai/memory/MemoryModuleType;)Lcom/google/common/collect/ImmutableMap;
+		ARG 1 memoryToAcquire
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/AcquirePoi.mapping:17]: The order is:
``memoryKey`` & ``memoryToAcquire`` i think.
Either way the second value is ``memoryToAcquire`` not the first, see the call to super in AcquirePoi constructor.

File: data/net/minecraft/world/entity/ai/behavior/GateBehavior.mapping [hunk 4]
```diff
@@ -17,3 +22,9 @@ CLASS net/minecraft/world/entity/ai/behavior/GateBehavior
 		ARG 3 gameTime
 	METHOD timedOut (J)Z
 		ARG 1 gameTime
+	CLASS OrderPolicy
+		METHOD <init> (Ljava/lang/String;ILjava/util/function/Consumer;)V
+			ARG 3 consumer
+	CLASS RunningPolicy
+		METHOD apply (Ljava/util/stream/Stream;Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/entity/LivingEntity;J)V
+			ARG 2 level
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/GateBehavior.mapping:31]: This should be in theory ``behaviour, level, owner, gameTime``

File: data/net/minecraft/world/entity/ai/behavior/GoToWantedItem.mapping [hunk 5]
```diff
@@ -1,4 +1,11 @@
 CLASS net/minecraft/world/entity/ai/behavior/GoToWantedItem
+	METHOD <init> (FZI)V
+		ARG 1 speedModifier
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/GoToWantedItem.mapping:3]: Arg 2 should be ``hasTarget``

File: data/net/minecraft/world/entity/ai/behavior/PlayTagWithOtherKids.mapping [hunk 6]
```diff
@@ -1,4 +1,6 @@
 CLASS net/minecraft/world/entity/ai/behavior/PlayTagWithOtherKids
+	METHOD chaseKid (Lnet/minecraft/world/entity/PathfinderMob;Lnet/minecraft/world/entity/LivingEntity;)V
+		ARG 1 interactionTarget
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/PlayTagWithOtherKids.mapping:4]: arg 1 is entity or memoryHolder
arg 2 is interactionTarget

File: data/net/minecraft/world/entity/ai/behavior/RunIf.mapping [hunk 7]
```diff
@@ -1,4 +1,17 @@
 CLASS net/minecraft/world/entity/ai/behavior/RunIf
+	METHOD <init> (Ljava/util/Map;Ljava/util/function/Predicate;Lnet/minecraft/world/entity/ai/behavior/Behavior;Z)V
+		ARG 2 predicate
+		ARG 3 wrappedBehavior
+		ARG 4 checkWhileRunningAlso
+	METHOD <init> (Ljava/util/Map;Lnet/minecraft/world/entity/ai/behavior/Behavior;)V
+		ARG 2 wrappedBehavior
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/RunIf.mapping:8]: arg 1 is entryConditions

File: data/net/minecraft/world/entity/ai/behavior/RunSometimes.mapping [hunk 8]
```diff
@@ -1,11 +1,19 @@
 CLASS net/minecraft/world/entity/ai/behavior/RunSometimes
+	METHOD <init> (Lnet/minecraft/world/entity/ai/behavior/Behavior;Lnet/minecraft/util/valueproviders/UniformInt;)V
+		ARG 1 wrappedBehavior
+		ARG 2 interval
+	METHOD <init> (Lnet/minecraft/world/entity/ai/behavior/Behavior;ZLnet/minecraft/util/valueproviders/UniformInt;)V
+		ARG 1 wrappedBehavior
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/RunSometimes.mapping:6]: arg2 should be ``keepTicks``

File: data/net/minecraft/world/entity/ai/behavior/SetHiddenState.mapping [hunk 9]
```diff
@@ -1,4 +1,7 @@
 CLASS net/minecraft/world/entity/ai/behavior/SetHiddenState
+	METHOD <init> (II)V
+		ARG 1 stayHiddenSeconds
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/SetHiddenState.mapping:3]: ``stayHiddenSeconds`` or ``durationInSeconds`` ???
@alcatrazEscapee @ChampionAsh5357

File: data/net/minecraft/world/entity/ai/behavior/SetWalkTargetAwayFrom.mapping [hunk 10]
```diff
@@ -1,7 +1,20 @@
 CLASS net/minecraft/world/entity/ai/behavior/SetWalkTargetAwayFrom
+	METHOD <init> (Lnet/minecraft/world/entity/ai/memory/MemoryModuleType;FIZLjava/util/function/Function;)V
+		ARG 1 walkTargetAwayFromMemory
+		ARG 2 speedModifier
+		ARG 3 desiredDistance
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/SetWalkTargetAwayFrom.mapping:5]: arg 4 is "hasTarget"

File: data/net/minecraft/world/entity/ai/behavior/SetWalkTargetAwayFrom.mapping [hunk 11]
```diff
@@ -1,7 +1,20 @@
...
+		ARG 5 toPosition
 	METHOD checkExtraStartConditions (Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/entity/PathfinderMob;)Z
 		ARG 1 level
 		ARG 2 owner
+	METHOD entity (Lnet/minecraft/world/entity/ai/memory/MemoryModuleType;FIZ)Lnet/minecraft/world/entity/ai/behavior/SetWalkTargetAwayFrom;
+		ARG 0 walkTargetAwayFromMemory
+		ARG 1 speedModifier
+		ARG 2 desiredDistance
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/SetWalkTargetAwayFrom.mapping:14]: arg 3 is "hasTarget"

File: data/net/minecraft/world/entity/ai/behavior/SetWalkTargetAwayFrom.mapping [hunk 12]
```diff
@@ -1,7 +1,20 @@
...
+	METHOD pos (Lnet/minecraft/world/entity/ai/memory/MemoryModuleType;FIZ)Lnet/minecraft/world/entity/ai/behavior/SetWalkTargetAwayFrom;
+		ARG 0 walkTargetAwayFromMemory
+		ARG 1 speedModifier
+		ARG 2 desiredDistance
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/SetWalkTargetAwayFrom.mapping:19]: arg 3 is "hasTarget"

File: data/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.mapping [hunk 13]
```diff
@@ -1,5 +1,17 @@
 CLASS net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory
+	METHOD <init> (Lnet/minecraft/world/entity/ai/memory/MemoryModuleType;FIII)V
+		ARG 1 memoryType
+		ARG 2 speedModifier
+		ARG 3 closeEnoughDist
+		ARG 4 tooFarDistance
+		ARG 5 tooLongUnreachableDuration
+	METHOD closeEnough (Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/entity/npc/Villager;Lnet/minecraft/core/GlobalPos;)Z
+		ARG 1 level
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.mapping:9]: Order of arguments are: ``level, villager, memoryPos``

File: data/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.mapping [hunk 14]
```diff
@@ -1,5 +1,17 @@
...
 	METHOD start (Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/entity/npc/Villager;J)V
 		ARG 1 level
 		ARG 2 entity
 		ARG 3 gameTime
+	METHOD tiredOfTryingToFindTarget (Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/entity/npc/Villager;)Z
+		ARG 1 level
+	METHOD wrongDimension (Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/core/GlobalPos;)Z
+		ARG 1 level
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.mapping:23]: Order of arguments are: ``level, memoryPos``

File: data/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.mapping [hunk 15]
```diff
@@ -1,5 +1,17 @@
...
+		ARG 4 tooFarDistance
+		ARG 5 tooLongUnreachableDuration
+	METHOD closeEnough (Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/entity/npc/Villager;Lnet/minecraft/core/GlobalPos;)Z
+		ARG 1 level
 	METHOD start (Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/entity/npc/Villager;J)V
 		ARG 1 level
 		ARG 2 entity
 		ARG 3 gameTime
+	METHOD tiredOfTryingToFindTarget (Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/entity/npc/Villager;)Z
+		ARG 1 level
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/SetWalkTargetFromBlockMemory.mapping:17]: Order of arguments are: ``level, villager``

File: data/net/minecraft/world/entity/ai/behavior/ValidateNearbyPoi.mapping [hunk 16]
```diff
@@ -1,7 +1,16 @@
 CLASS net/minecraft/world/entity/ai/behavior/ValidateNearbyPoi
+	METHOD <init> (Lnet/minecraft/world/entity/ai/village/poi/PoiType;Lnet/minecraft/world/entity/ai/memory/MemoryModuleType;)V
+		ARG 1 poiType
+		ARG 2 memoryType
+	METHOD bedIsOccupied (Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/entity/LivingEntity;)Z
+		ARG 1 level
+		ARG 2 pos
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/ValidateNearbyPoi.mapping:7]: Order of arguments are: ``level, pos, sleeper``

File: data/net/minecraft/world/entity/ai/behavior/WorkAtComposter.mapping [hunk 17]
```diff
@@ -5,6 +5,8 @@ CLASS net/minecraft/world/entity/ai/behavior/WorkAtComposter
 		ARG 4 state
 	METHOD makeBread (Lnet/minecraft/world/entity/npc/Villager;)V
 		ARG 1 villager
+	METHOD spawnComposterFillEffects (Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/state/BlockState;)V
+		ARG 1 level
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/ai/behavior/WorkAtComposter.mapping:9]: Order of arguments are: ``level, preState, pos, postState``

File: data/net/minecraft/world/entity/animal/Rabbit.mapping [hunk 18]
```diff
@@ -29,6 +29,8 @@ CLASS net/minecraft/world/entity/animal/Rabbit
 		ARG 1 stack
 	METHOD isTemptingItem (Lnet/minecraft/world/item/Item;)Z
 		ARG 1 item
+	METHOD isTemptingItem (Lnet/minecraft/world/item/ItemStack;)Z
+		ARG 0 item
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/animal/Rabbit.mapping]: ``item`` -> ``stack``

File: data/net/minecraft/world/entity/animal/TropicalFish.mapping [hunk 19]
```diff
@@ -5,6 +5,8 @@ CLASS net/minecraft/world/entity/animal/TropicalFish
 		ARG 0 pattern
 		ARG 1 baseColor
 		ARG 2 patternColor
+	METHOD checkTropicalFishSpawnRules (Lnet/minecraft/world/entity/EntityType;Lnet/minecraft/world/level/LevelAccessor;Lnet/minecraft/world/entity/MobSpawnType;Lnet/minecraft/core/BlockPos;Ljava/util/Random;)Z
+		ARG 4 random
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/animal/TropicalFish.mapping:9]: Order of arguments are: ``type, level, spawnType, pos, random``

File: data/net/minecraft/world/entity/animal/Wolf.mapping [hunk 20]
```diff
@@ -6,6 +6,8 @@ CLASS net/minecraft/world/entity/animal/Wolf
 	METHOD canMate (Lnet/minecraft/world/entity/animal/Animal;)Z
 		COMMENT Returns true if the mob is currently able to mate with the specified mob.
 		ARG 1 otherAnimal
+	METHOD checkWolfSpawnRules (Lnet/minecraft/world/entity/EntityType;Lnet/minecraft/world/level/LevelAccessor;Lnet/minecraft/world/entity/MobSpawnType;Lnet/minecraft/core/BlockPos;Ljava/util/Random;)Z
+		ARG 4 random
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/animal/Wolf.mapping:14]: Order of arguments are: ``type, level, spawnType, pos, random``

File: data/net/minecraft/world/entity/animal/goat/Goat.mapping [hunk 21]
```diff
@@ -0,0 +1,3 @@
+CLASS net/minecraft/world/entity/animal/goat/Goat
+	METHOD setScreamingGoat (Z)V
+		ARG 1 screamingGoat
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/animal/goat/Goat.mapping]: ``screamingGoat`` -> ``isScreamingGoat``

File: data/net/minecraft/world/entity/monster/Skeleton.mapping [hunk 22]
```diff
@@ -5,3 +5,7 @@ CLASS net/minecraft/world/entity/monster/Skeleton
 		ARG 3 recentlyHit
 	METHOD getHurtSound (Lnet/minecraft/world/damagesource/DamageSource;)Lnet/minecraft/sounds/SoundEvent;
 		ARG 1 damageSource
+	METHOD setFreezeConverting (Z)V
+		ARG 1 freezeConverting
```
Comment (Lanse505) [inline data/net/minecraft/world/entity/monster/Skeleton.mapping]: ``freezeConverting`` -> ``isFrozen``

**Review by Lanse505 — CHANGES_REQUESTED** [review]
Overall looks good, some changes need to be done before it can be merged though.
There is a singular javadoc change, not too experienced with our javadocs policy so I'll defer it to @alcatrazEscapee @sciwhiz12 or @ChampionAsh5357

File: data/net/minecraft/world/entity/ai/behavior/AcquirePoi.mapping (hunk 3 above)
Comment (PiTheGuy) [reply data/net/minecraft/world/entity/ai/behavior/AcquirePoi.mapping:17]: I'll add `memoryKey` as the name for the first parameter, but in Enigma it already shows `memoryToAcquire` as the second parameter, not the first.

File: data/net/minecraft/world/entity/AgeableMob.mapping (hunk 1 above)
Comment (PiTheGuy) [reply data/net/minecraft/world/entity/AgeableMob.mapping:3]: Changed

File: data/net/minecraft/world/entity/ai/behavior/AcquirePoi.mapping (hunk 2 above)
Comment (PiTheGuy) [reply data/net/minecraft/world/entity/ai/behavior/AcquirePoi.mapping]: Fixed

**Comment by PiTheGuy** [comment]
All requested changes should have been applied. The only exception is the one waiting for a second opinion from members of the team.

File: data/net/minecraft/world/entity/ai/behavior/SetHiddenState.mapping (hunk 9 above)
Comment (alcatrazEscapee) [reply data/net/minecraft/world/entity/ai/behavior/SetHiddenState.mapping:3]: Mojang field is `stayHiddenTicks`, the parameter is multiplied by 20 and assigned, I think `stayHiddenSeconds` makes sense here.

File: data/net/minecraft/world/entity/player/Inventory.mapping [hunk 23]
```diff
@@ -33,7 +33,7 @@ CLASS net/minecraft/world/entity/player/Inventory
 	METHOD findSlotMatchingUnusedItem (Lnet/minecraft/world/item/ItemStack;)I
 		ARG 1 stack
 	METHOD getArmor (I)Lnet/minecraft/world/item/ItemStack;
-		COMMENT returns a player armor item (as itemstack) contained in specified armor slot.
+		COMMENT @return a player armor item (as itemstack) contained in specified armor slot.
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/entity/player/Inventory.mapping]:
```suggestion
		COMMENT @return a player armor item (as an {@code ItemStack}) contained in specified armor slot
```

Generally for formatting, we allow for `@return` and `@param`, any of the following styles:
 - `@return the foo`
 - `@return the foo. And another full sentence.`
 - `@return The Foo. And another full sentence.`

**Review by Lanse505 — APPROVED** [review]
Looks good to me

**Pull request merged**